tools/remove_dependencies.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+# Copyright 2023 The Vitess Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Remove tools and dependencies installed by "make tools"
+
+set -euo pipefail
+
+function fail() {
+	echo "ERROR: ${1}"
+	exit 1
+}
+
+BUILD_JAVA=${BUILD_JAVA:-1}
+BUILD_CONSUL=${BUILD_CONSUL:-1}
+uname=$(uname)
+get_arch=$(uname -m)
+
+uninstall_protoc() {
+    echo "Removing protoc..."
+    local dist="$1"
+
+    if [ -f "$dist/bin/protoc" ]; then
+        unlink "$dist/bin/protoc"
+        rm "$VTROOT/bin/protoc"
+    fi
+    if [[ "$(echo $dist | awk -F/ '{print $NF}')" == "vt-protoc-$PROTOC_VER" ]]; then
```

Comment:
This method is fine, but you can also do this more efficiently in bash this way:
```
if [[ ${dir##*/} == "vt-protoc-$PROTOC_VER" ]]; then
```

This also removes an external dependency (awk isn't quite the same between the BSD and GNU toolchains e.g.).

Reply:
[basename](https://linux.die.net/man/1/basename) is also an option.

Reply:
@mattlord Should I implement `basename`? It's been a long time since I opened this PR.

Reply:
Any reason not to use the bash feature shown? That will be the most efficient thing and eliminates an unnecessary external dependency for the script. It also looks like we'll need to merge in origin/main to address some test failures as well.

Reply:
No specific reason as such. Basename is also widely supported. We can go with your bash approach.